generated/proto/encodingpb/time.proto

```diff
@@ -0,0 +1,23 @@
+syntax = "proto3";
+
+option go_package = "github.com/xichen2020/eventdb/generated/proto/encodingpb";
+
+import "github.com/xichen2020/eventdb/generated/proto/encodingpb/types.proto";
+
+package encodingpb;
+
+enum ResolutionType {
+    UNKNOWN_RESOLUTION = 0;
+    NANOSECOND = 1;
+    MICROSECOND = 2;
+    MILLISECOND = 3;
+    SECOND = 4;
+}
+
+message TimeMeta {
+    EncodingType encoding = 1;
+    ResolutionType resolution = 2;
+    // Bits per encoded value. All encoded values are bit packed.
+    int64 bits_per_encoded_value = 3;
+    int64 delta_start = 4;
```

Comment:
Hm why is this in the proto? Shouldnit it be encoded together with the data stream?

Reply:
This is to encode the first value so that we can store the first value as 0. 

Since we are only reserving enough bits to store the max delta, in some cases we won't have enough bits to store the full first value.